src/data/sym/basic.lean

```diff
@@ -158,46 +158,62 @@ end inhabited
 instance has_zero : has_zero (sym α 0) := ⟨⟨0, rfl⟩⟩
 instance has_emptyc : has_emptyc (sym α 0) := ⟨0⟩
 
+lemma eq_nil_of_card_zero (s : sym α 0) : s = nil :=
+begin
+  obtain ⟨a, h⟩ := s,
+  rw multiset.card_eq_zero at h,
+  subst h,
+  refl,
+end
+
+instance unique_zero : unique (sym α 0) :=
+⟨⟨nil⟩, eq_nil_of_card_zero⟩
+
+/-- `repeat a n` is the sym containing only `a` with multiplicity `n`. -/
+def repeat (a : α) (n : ℕ) : sym α n := ⟨multiset.repeat a n, multiset.card_repeat _ _⟩
+
+lemma repeat_succ {a : α} {n : ℕ} : repeat a n.succ = a :: repeat a n := rfl
+
+lemma exists_eq_cons_of_succ (s : sym α n.succ) : ∃ (a : α) (s' : sym α n), s = a :: s' :=
+begin
+  classical,
+  obtain ⟨m, h⟩ := s,
+  have : 0 < multiset.card m,
+  { rw h,
+    exact nat.succ_pos n, },
+  rw multiset.card_pos_iff_exists_mem at this,
+  obtain ⟨a, ha⟩ := this,
+  use [a, m.erase a],
+  { rw [multiset.card_erase_of_mem ha, h],
+    refl, },
+  { rw cons,
+    congr,
+    rw multiset.cons_erase ha, },
+end
+
+lemma eq_repeat_of_subsingleton [subsingleton α] (a : α) {n : ℕ} (s : sym α n) : s = repeat a n :=
+begin
+  induction n with n ih,
+  { rw eq_nil_of_card_zero s,
+    refl, },
+  { obtain ⟨b, s', rfl⟩ := exists_eq_cons_of_succ s,
+    rw [repeat_succ, ih s'],
+    congr, },
+end
+
 instance subsingleton (n : ℕ) [subsingleton α] : subsingleton (sym α n) :=
 ⟨begin
-  rintros ⟨a, ha⟩ ⟨b, hb⟩,
-  simp only [subtype.mk.inj_eq],
-  induction a using multiset.case_strong_induction_on with k hk ih generalizing n b,
-  { rw [ha.symm, multiset.card_zero] at hb, exact (multiset.card_eq_zero.mp hb).symm, },
-  { cases n,
-    { exact (multiset.cons_ne_zero (multiset.card_eq_zero.mp ha)).elim },
-    { by_cases hzero : b = 0,
-      { rw [hzero, multiset.card_zero] at hb,
-        exact (n.succ_ne_zero hb.symm).elim },
-      { have hmem := multiset.exists_mem_of_ne_zero hzero,
-        rcases hmem with ⟨r, hr⟩,
-        cases multiset.exists_cons_of_mem hr,
-        rw h,
-        have heq := @ih hk rfl.ge n w begin
-          rw multiset.card_cons at ha,
-          refine nat.succ.inj ha,
-        end begin
-          rw [h, multiset.card_cons] at hb,
-          refine nat.succ.inj hb,
-        end,
-        rw [heq, subsingleton.elim k r] } } }
+  cases n,
+  { simp, },
+  { intros s s',
+    obtain ⟨b, s, rfl⟩ := exists_eq_cons_of_succ s,
+    rw [eq_repeat_of_subsingleton b s', eq_repeat_of_subsingleton b s, repeat_succ], },
 end⟩
 
-instance unique (n : ℕ) [unique α] : unique (sym α n) := unique.mk' _
-
 instance is_empty (n : ℕ) [is_empty α] : is_empty (sym α n.succ) :=
```

Comment:
You usually don't have to name instances. Here, Lean will name it `sym.is_unique`, as expected.
```suggestion
instance (n : ℕ) [is_empty α] : is_empty (sym α n.succ) :=
```